src/main/io/gps.c

```diff
@@ -106,12 +106,26 @@ typedef struct gpsInitData_s {
 
 // NMEA will cycle through these until valid data is received
 static const gpsInitData_t gpsInitData[] = {
+#ifdef USE_GPS_MTK
     { GPS_BAUDRATE_115200,   BAUD_115200, "$PUBX,41,1,0003,0001,115200,0*1E\r\n", "$PMTK251,115200*1F\r\n" },
     { GPS_BAUDRATE_57600,    BAUD_57600,  "$PUBX,41,1,0003,0001,57600,0*2D\r\n",  "$PMTK251,57600*2C\r\n" },
     { GPS_BAUDRATE_38400,    BAUD_38400,  "$PUBX,41,1,0003,0001,38400,0*26\r\n",  "$PMTK251,38400*27\r\n" },
     { GPS_BAUDRATE_19200,    BAUD_19200,  "$PUBX,41,1,0003,0001,19200,0*23\r\n",  "$PMTK251,19200*22\r\n" },
     // 9600 is not enough for 5Hz updates - leave for compatibility to dumb NMEA that only runs at this speed
     { GPS_BAUDRATE_9600,     BAUD_9600,   "$PUBX,41,1,0003,0001,9600,0*16\r\n",   "" }
+#elif defined(USE_GPS_SIRF)
+    { GPS_BAUDRATE_115200,   BAUD_115200, "$PUBX,41,1,0003,0001,115200,0*1E\r\n", "$PSRF100,0,115200,8,1,0*04\r\n" },
+    { GPS_BAUDRATE_57600,    BAUD_57600,  "$PUBX,41,1,0003,0001,57600,0*2D\r\n",  "$PSRF100,0,57600,8,1,0*37\r\n" },
+    { GPS_BAUDRATE_38400,    BAUD_38400,  "$PUBX,41,1,0003,0001,38400,0*26\r\n",  "$PSRF100,0,38400,8,1,0*3C\r\n" },
+    { GPS_BAUDRATE_19200,    BAUD_19200,  "$PUBX,41,1,0003,0001,19200,0*23\r\n",  "$PSRF100,0,19200,8,1,0*39\r\n" },
+    { GPS_BAUDRATE_9600,     BAUD_9600,   "$PUBX,41,1,0003,0001,9600,0*16\r\n",   "$PSRF100,0,9600,8,1,0*0C\r\n" }
+#else
+    { GPS_BAUDRATE_115200,   BAUD_115200, "$PUBX,41,1,0003,0001,115200,0*1E\r\n", "$PSRF100,1,115200,8,1,0*05\r\n" },
+    { GPS_BAUDRATE_57600,    BAUD_57600,  "$PUBX,41,1,0003,0001,57600,0*2D\r\n",  "$PSRF100,1,57600,8,1,0*36\r\n" },
+    { GPS_BAUDRATE_38400,    BAUD_38400,  "$PUBX,41,1,0003,0001,38400,0*26\r\n",  "$PSRF100,1,38400,8,1,0*3D\r\n" },
+    { GPS_BAUDRATE_19200,    BAUD_19200,  "$PUBX,41,1,0003,0001,19200,0*23\r\n",  "$PSRF100,1,19200,8,1,0*38\r\n" },
+    { GPS_BAUDRATE_9600,     BAUD_9600,   "$PUBX,41,1,0003,0001,9600,0*16\r\n",   "$PSRF100,1,9600,8,1,0*0D\r\n" }
+#endif
```

Comment:
Should the final else on this be the proprietary u-blox commands?

In this struct:
```
typedef struct gpsInitData_s {
    uint8_t index;
    uint8_t baudrateIndex; // see baudRate_e
    const char *ubx;
    const char *mtk;
} gpsInitData_t;
```
That last element `mtk` should probably be renamed to `nmea` since this is the string we want to send when we have `set gps_provider = NMEA`. With these changes, it won't just be a proprietary GlobalTop Tech (MTK) command, it could one of the three.

Having the proprietary SiRF Tech commands be used by default seems like it will break a lot of GPS modules people will be using, which will mostly be u-blox based hardware.

The string would be the same as the `.ubx` string, since we use the NMEA port initialization command when in UBX protocol.

Reply:
We set NMEA message mode by default.

`SiRF` will only be used when adding custom define `GPS_SIRF`
Old `MTK` commands will be used when adding custom define `GPS_MTK`.

Using `.nmea` attribute now

Reply:
The final else in that block loads SiRF commands ($PSRF...) into the `.nmea` element, doesn't it?

So if we don't define either of those, we will still be using SiRF commands.
My suggestion is by default to use u-blox proprietary commands, since I think most users will have u-blox based modules.

Reply:
Using ubx as default